### DISCUSSION TITLE
Sourcing an R script for an entire Quarto project

### DISCUSSION THREAD

**Discussion by arnaudgallou** [quarto-dev/quarto-cli#10491] [Q&A] [answered]
### Description

I'd like to import/source an R script that contains helper functions for an entire Quarto project. Sourcing a script in a code chunk at the top level of `index.qmd` doesn't work for complex project such as books that requires sourcing a particular script in every `.qmd` file. See [here](https://github.com/arnaudgallou/test-qp) for dummy project that will fail on rendering with the following error:

```
Quitting from lines 4-5 [unnamed-chunk-1] (intro.qmd)
Error in `bar()`:
! could not find function "bar"
```

Tested on macOS 13.6.7 and Quarto v. 1.5.56.

Is there a way to import a script for an entire project? I was hoping for something similar to `pre-render` that will run during the rendering process but that doesn't seem to exist.

**Comment by mcanouil** [comment, marked as answer]
No, you need to load it in every document.
Note that to avoid duplicating the code, you can put the code cell in a Quarto document and use the include shortcode in every document.
Each Quarto documents are computed independently.

This has already been discussed:

- https://github.com/quarto-dev/quarto-cli/discussions/3250
